frontend/express/public/javascripts/countly/countly.vue.components.js

```diff
@@ -372,4 +373,2886 @@
+                }
+            },
+            watch: {
+                dataPoints: function() {
+                    this.refresh();
+                },
+                graphType: function() {
+                    this.refresh();
+                },
+                frozen: function(newValue) {
+                    if (!newValue) {
+                        this.refresh();
+                    }
+                }
+            },
+            mounted: function() {
+                this.refresh();
+            },
+            methods: {
+                refresh: function() {
+
+                    if (this.frozen || $(this.$refs.container).is(":hidden") || !this.hasData) {
+                        // no need to refresh if hidden
+                        return;
+                    }
+
+                    countlyCommon.drawGraph(this.dataPoints,
+                        $(this.$refs.container),
+                        this.graphType,
+                        this.options);
+                }
+            },
+            template: '<div class="cly-vue-graph-w">\
+                            <div ref="container" class="graph-container"></div>\
+                            <div class="cly-vue-graph-no-data" v-if="!hasData">\
+                                <div class="inner">\
+                                    <div class="icon"></div>\
+                                    <div class="text">{{i18n("common.graph.no-data")}}</div>\
+                                </div>\
+                            </div>\
+                        </div>'
+        }
+    ));
+
+    Vue.component("cly-radio", countlyBaseComponent.extend(
+        // @vue/component
+        {
+            props: {
+                value: {required: true, default: -1, type: [ String, Number ]},
+                items: {
+                    required: true,
+                    type: Array,
+                    default: function() {
+                        return [];
+                    }
+                },
+                skin: { default: "main", type: String}
+            },
+            computed: {
+                skinClass: function() {
+                    if (["main", "light"].indexOf(this.skin) > -1) {
+                        return "radio-" + this.skin + "-skin";
+                    }
+                    return "radio-main-skin";
+                }
+            },
+            methods: {
+                setValue: function(e) {
+                    this.$emit('input', e);
+                }
+            },
+            template: '<div class="cly-vue-radio" v-bind:class="[skinClass]">\
+                            <div class="radio-wrapper">\
+                                <div @click="setValue(item.value)" v-for="(item, i) in items" :key="i" :class="{\'selected\': value == item.value}" class="radio-button">\
+                                    <div class="box"></div>\
+                                    <div class="text">{{item.label}}</div>\
```

Comment:
Codacy found an issue: [Bad escaping of EOL. Use option multistr if needed.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=6339769)